march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -257,3 +269,24 @@ void HardwareBuilder::initUrdf()
     this->init_urdf_ = false;
   }
 }
+
+std::string rightHandJustifyString(std::string input)
+{
+  std::string delimiter = "\n";
+  if (input.length() > delimiter.length())
+  {
+    while (input.size() < 4 + delimiter.length())  // check until the 16-bit number has been filled
+    {
+      input.insert(0, "0");
+    }
+  }
+  return input;
+}
+
+std::stringstream convertSWFileToStringStream(std::ifstream& sw_file)
+{
+  std::stringstream sw_stream;
+  std::copy(std::istreambuf_iterator<char>(sw_file), std::istreambuf_iterator<char>(),
+            std::ostreambuf_iterator<char>(sw_stream));
+  return sw_stream;
+}
```

Comment:
See my above comment about using a string instead of a stream. You could change this method like so
```suggestion
std::string convertSWFileToString(std::ifstream& sw_file)
{
  return std::string(std::istreambuf_iterator<char>(sw_file),
                     std::istreambuf_iterator<char>());
}
```